src/go/wsl-helper/cmd/kubeconfig.go

```diff
@@ -57,65 +61,101 @@ var kubeconfigCmd = &cobra.Command{
 		cmd.SilenceUsage = true
 
 		configDir := path.Join(homedir.HomeDir(), ".kube")
-		linkPath := path.Join(configDir, "config")
-		if show {
-			// The output is "true", "false", or an error message for UI.
-			// We will only return nil in this path.
-			target, err := os.Readlink(linkPath)
-			if err != nil {
-				if errors.Is(err, os.ErrNotExist) {
-					fmt.Println("false")
-				} else if errors.Is(err, syscall.EINVAL) {
-					fmt.Printf("File %s exists and is not a symlink\n", linkPath)
-				} else {
-					fmt.Printf("%s\n", err)
-				}
-			} else if target == configPath {
-				fmt.Println("true")
-			} else {
-				// For a symlink pointing elsewhere, we assume we can overwrite.
-				fmt.Println("false")
-			}
-			return nil
+
+		configFile, err := os.Open(configPath)
+		if err != nil {
+			return err
 		}
+
+		var config kubeConfig
+		defer configFile.Close()
+		err = yaml.NewDecoder(configFile).Decode(&config)
+		if err != nil {
+			return err
+		}
+
+		cleanConfig := removeExistingRDCluster(rdCluster, &config)
+
+		kubeConfig, err := updateClusterIP(*cleanConfig, rdNetworking)
+
+		var finalKubeConfigFile *os.File
 		if enable {
-			err = os.Mkdir(configDir, 0o750)
-			if err != nil && !errors.Is(err, os.ErrExist) {
-				// The error already contains the full path, we can't do better.
+			if err := os.MkdirAll(configDir, 0o750); err != nil {
 				return err
 			}
-			err = os.Symlink(configPath, linkPath)
+			finalKubeConfigFile, err = os.Create(filepath.Join(configDir, "config"))
 			if err != nil {
-				if errors.Is(err, os.ErrExist) {
-					// If it already exists, do nothing; even if it's not a symlink.
-					return nil
-				}
 				return err
 			}
-		} else {
-			// No need to create if we want to remove it
-			target, err := os.Readlink(linkPath)
-			if err != nil {
-				if errors.Is(err, os.ErrNotExist) {
-					return nil
-				}
+			defer finalKubeConfigFile.Close()
+			err = os.Mkdir(configDir, 0o750)
+			if err != nil && !errors.Is(err, os.ErrExist) {
+				// The error already contains the full path, we can't do better.
 				return err
 			}
-			if target == configPath {
-				err = os.Remove(linkPath)
-				if err != nil && !errors.Is(err, os.ErrNotExist) {
-					return err
-				}
+			err = yaml.NewEncoder(finalKubeConfigFile).Encode(kubeConfig)
+			if err != nil {
+				return err
 			}
 		}
 		return nil
 	},
 }
 
+func updateClusterIP(config kubeConfig, rdNetworking bool) (kubeConfig, error) {
+	ip, err := getClusterIP()
+	if err != nil {
+		return config, err
+	}
+	// Fix up any clusters at 127.0.0.1, using the IP address we found.
+	for clusterIdx, cluster := range config.Clusters {
```

Comment:
We're reading the _Windows_ kubeconfig here, so `.cluster.name` should already be `rancher-desktop`. That should make things easier to implement…